datasource/common/src/main/java/org/opennms/alec/datasource/common/util/AlarmUtil.java

```diff
@@ -0,0 +1,30 @@
+package org.opennms.alec.datasource.common.util;
+
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import org.opennms.alec.datasource.api.Alarm;
+
+public class AlarmUtil {
+
+    public static Alarm getAlarmForDescription(final Collection<Alarm> alarms) {
```

Comment:
Let's adapt this to choose the oldest alarm with the highest severity instead of the current logic.

So, the highest severity alarm should be chosen and if there are many with the highest severity, the oldest one will be picked.

Reply:
done.